selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/NewWorkspace.java

```diff
@@ -117,6 +150,88 @@ public NewWorkspace(
     String MULTI_MACHINE_BUTTON_ID = "multi-machine-button";
     String ADD_OR_IMPORT_PROJECT_BUTTON_ID = "ADD_PROJECT";
+    String ADD_OR_CANCEL_BUTTON_XPATH_TEMPLATE = "//*[@id='%s']/button";
+
+    // "Add or Import Project" form buttons
+    String SAMPLES_BUTTON_ID = "samples-button";
+    String BLANK_BUTTON_ID = "blank-button";
+    String GIT_BUTTON_ID = "git-button";
+    String GITHUG_BUTTON_ID = "github-button";
+    String ZIP_BUTTON_ID = "zip-button";
+    String ADD_BUTTON_ID = "add-project-button";
+    String CANCEL_BUTTON_ID = "cancel-button";
+
+    String CHECKBOX_BY_SAMPLE_NAME_ID_TEMPLATE = "sample-%s";
+    String PROJECT_TAB_XPATH_TEMPLATE = "//toggle-single-button[@id='%s']/div/button/div";
+    String PROJECT_TAB_BY_NAME_XPATH_TEMPLATE =
+        "//div[@class='%s']//span[text()='console-java-simple']";
+
+    // Project options
+    String CANCEL_PROJECT_OPTIONS_BUTTON_XPATH = "//button[@name='cancelButton']";
+    String SAVE_PROJECT_OPTIONS_BUTTON_XPATH = "//che-button-primary//button[@name='saveButton']";
+    String PROJECT_NAME_ERROR_MESSAGE_XPATH =
+        "//div[@che-name='projectName']//div[@id='new-workspace-error-message']/div";
+    String REPOSITORY_URL_ERROR_MESSAGE_XPATH =
+        "//div[@che-name='projectGitURL']//div[@id='new-workspace-error-message']/div";
+
+    // Blank tab fields
+    String BLANK_FORM_NAME_FIELD_XPATH = "//input[@name='name']";
+    String BLANK_FORM_DESCRIPTION_FIELD_XPATH = "//input[@name='description']";
+    String BLANK_FORM_NAME_ERROR_MESSAGE_XPATH =
+        "//div[@id='project-source-selector']//div[@id='new-workspace-error-message']/div";
+  }
+
+  public enum StackId {
+    BLANK("blank-default"),
+    JAVA("java-default"),
+    JAVA_MYSQL("java-mysql"),
+    DOT_NET("dotnet-default"),
+    ANDROID("android-default"),
+    CPP("cpp-default"),
+    CENTOS_BLANK("centos"),
+    CENTOS_GO("centos-go"),
+    CENTOS_NODEJS("nodejs4"),
+    CENTOS_WILDFLY_SWARM("wildfly-swarm"),
+    CENTOS_WITH_JAVA_JAVASCRIPT("ceylon-java-javascript-dart-centos"),
+    DEBIAN("debian"),
+    DEBIAN_LSP("debianlsp"),
+    ECLIPSE_CHE("che-in-che"),
+    ECLIPSE_VERTX("vert.x"),
+    GO("go-default"),
+    HADOOP("hadoop-default"),
+    JAVA_CENTOS("java-centos"),
+    JAVA_DEBIAN("java-debian"),
+    JAVA_THEIA_DOCKER("java-theia-docker"),
+    JAVA_THEIA_OPENSHIFT("java-theia-openshift"),
+    JAVA_MYSQL_CENTOS("java-centos-mysql"),
+    KOTLIN("kotlin-default"),
+    NODE("node-default"),
+    OPENSHIFT_DEFAULT("openshift-default"),
+    OPENSHIFT_SQL("openshift-sql"),
+    PHP("php-default"),
+    PHP_GAE("php-gae"),
+    PHP_5_6("php5.6-default"),
+    PLATFORMIO("platformio"),
+    PYTHON("python-default"),
+    PYTHON_2_7("python-2.7"),
+    PYTHON_GAE("python-gae"),
+    RAILS("rails-default"),
+    SELENIUM("selenium"),
+    SPRING_BOOT("spring-boot"),
+    TOMEE("tomee-default"),
+    UBUNTU("ubuntu"),
+    ZEND("zend");
+
+    private final String stackId;
```

Comment:
Commonly, when we need to hold some value as enum attribute, the name of the attribute is differ from enum class name, in this case as follow:
```
public enum Stack {
  BLANK("blank-default"),
  ...;
 
  private final String id;

  Stack(String id) {
     this.id = id;
  }

   public String getId() {
     return id;
   }
}

```